napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1568 @@
+        self._moving_shape = None
+        self._moving_vertex = None
+        self._hover_shape = None
+        self._hover_vertex = None
+        if self._is_creating is True and self.mode == Mode.AddPath:
+            vertices = self.data._vertices[self.data._index == index]
+            if len(vertices) <= 2:
+                self.data.remove(index)
+            else:
+                self.data.edit(index, vertices[:-1])
+        if self._is_creating is True and self.mode == Mode.AddPolygon:
+            vertices = self.data._vertices[self.data._index == index]
+            if len(vertices) <= 2:
+                self.data.remove(index)
+        self._is_creating = False
+        self.refresh()
+
+    def remove_selected(self):
+        """Remove any selected shapes.
+        """
+        to_remove = sorted(self.selected_shapes, reverse=True)
+        for index in to_remove:
+            self.data.remove(index)
+        self.selected_shapes = []
+        shape, vertex = self._shape_at(self._cursor_coord)
+        self._hover_shape = shape
+        self._hover_vertex = vertex
+        self.status = self.get_message(self._cursor_coord, shape, vertex)
+        self.refresh()
+
+    def _rotate_box(self, angle, center=[0, 0]):
+        """Perfrom a rotation on the selected box.
+
+        Parameters
+        ----------
+        angle : float
+            angle specifying rotation of shapes in degrees.
+        center : list
+            coordinates of center of rotation.
+        """
+        theta = np.radians(angle)
+        transform = np.array([[np.cos(theta), np.sin(theta)],
+                             [-np.sin(theta), np.cos(theta)]])
+        box = self._selected_box - center
+        self._selected_box = box @ transform.T + center
+
+    def _scale_box(self, scale, center=[0, 0]):
+        """Perfrom a scaling on the selected box.
+
+        Parameters
+        ----------
+        scale : float, list
+            scalar or list specifying rescaling of shape.
+        center : list
+            coordinates of center of rotation.
+        """
+        if not isinstance(scale, (list, np.ndarray)):
+            scale = [scale, scale]
+        box = self._selected_box - center
+        box = np.array(box*scale)
+        if not np.all(box[1] == box[9]):
+            rescale = self._get_rescale()
+            r = self._rotation_handle_length*rescale
+            box[9] = box[1] + r*(box[9]-box[1])/np.linalg.norm(box[9]-box[1])
+        self._selected_box = box + center
+
+    def _transform_box(self, transform, center=[0, 0]):
+        """Perfrom a linear transformation on the selected box.
+
+        Parameters
+        ----------
+        transform : np.ndarray
+            2x2 array specifying linear transform.
+        center : list
+            coordinates of center of rotation.
+        """
+        box = self._selected_box - center
+        box = box @ transform.T
+        if not np.all(box[1] == box[9]):
```

Comment:
You have a lot of these integer constants all over the place. It might be helpful if you created a file called _constants, where you could store them all with names like BOX_INDEX=9 or whatever you want to name it. Then throughout the code when you want that number just use BOX_INDEX. This allows you to keep all the constants straight and define all of them in one place so you don't run into bugs where you change it in in spot but forget in another (you'd only have to change the value of the constant in the constants file)

Reply:
I have created a `_constants.py` file and added these